### DISCUSSION TITLE
docs: how to autoLoadEntities "base entities" when tree shaking?

### DISCUSSION THREAD

**Discussion by xenoterracide** [mikro-orm/mikro-orm#3724] [Q&A] [answered]
> Note that entities that aren't registered through the forFeature() method, but are only referenced from the entity (via a relationship), won't be included by way of the autoLoadEntities setting.

> You should not register your base entities via forFeature(), as there are no repositories for those. On the other hand, base entities need to be part of the list in forRoot() (or in the ORM config in general).

and of course it says that `forFeature()` is there so that you aren't bleeding these things outside your module. So how are you supposed to load entities that are referenced via relationships but don't have their own repositories (part of the aggregate but not an aggregate root). Note: I know this seems like a question, but I think it would be best answered in documentation.

**Comment by B4nan** [comment, marked as answer]
I don't understand your question. 

- There are no base entities during runtime (not for the ORM/metadata), their props are inlined to the implementations. 
- Every normal/leaf entity has a repository available.
- How is this about tree shaking? The base classes are extended in your entities, so part of the dependency graph when it comes to tree shaking. 
- Base entities should be discovered automatically in current version (although I am not 100% sure it works for multiple nested base classes).
- You can always combine the autoloading with explicit `entities` option in ORM config (should be fine for base entities).

Do you have an actual issue or is this just a confusion with the docs? I guess they are a bit dated.

**Reply by xenoterracide** [reply]
I was totally getting an issue where it was telling me the metadata wasn't there, to workaround I've had to add the other entities in forFeature even though it says not to because they wouldn't have a repo.

**Reply by B4nan** [reply]
What metadata? For base entity? There are no metadata on runtime for such, as said above, those are inlined. You can't target base entity in a relation.

**Reply by xenoterracide** [reply]
what's a "base entity"? maybe that's what I'm missing

**Reply by B4nan** [reply]
Entity you extend from?

**Reply by xenoterracide** [reply]
ah, no, I'm talking relationship entities, like ManyToOne. I think I understand now. The terminology "Base Entity" isn't one I've seen before.
